src/mesh.cc

```diff
@@ -1,96 +1,77 @@
-
 #include "mesh.h"
 
+void pipenetwork::Mesh::create_junctions(
+    const std::vector<Index>& ids, const std::vector<double>& elevations,
+    const std::vector<double>& demands,
+    const std::vector<double>& leak_diameters) {
 
-// Create nodal pointers and assign indices based on input coordinates
-void pipenetwork::Mesh::create_nodes(
-    const std::vector<Eigen::Vector3d>& coords) {
-  Index index = 0;
-  for (const auto& coord : coords) {
-    nodes_.emplace(std::pair<Index, std::shared_ptr<pipenetwork::Node>>(
-        index, std::make_shared<pipenetwork::Node>(index, coord)));
-    ++index;
+  Index i = 0;
+  for (const auto& id : ids) {
+    nodes_.emplace(id, std::make_shared<pipenetwork::Junction>(
```

Comment:
What happens if this fails?

Reply:
I guess our program will crash?  We dont want a partly built mesh to be solved, right?